### PR TITLE
core-files: Add missing directories and initialize them

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR fixes some strange errors during boot relating to mount failures.

### CODE REVIEW

**Review by avdgrinten — COMMENTED** [review]
Why is /boot/grub created unconditionally? Shouldn't grub create that at installation time? (And if GRUB is not installed, it should not be needed.)

**Comment by Dennisbonke** [comment]
I've just replicated the directories from update-image.py, it seems to be created unconditionally there too.

File: bootstrap.yml [hunk 1]
```diff
@@ -678,19 +678,36 @@ packages:
...
+      # xbps does not install empty directories, .keep files fixes this
+      - args: ['touch', '@THIS_COLLECT_DIR@/etc/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/run/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/root/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/dev/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/sys/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/proc/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/tmp/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/boot/grub/.keep']
+      - args: ['touch', '@THIS_COLLECT_DIR@/home/managarm/.keep']
```
Comment (ArsenArsen) [inline bootstrap.yml:709]: likewise

File: bootstrap.yml [hunk 2]
```diff
@@ -678,19 +678,36 @@ packages:
...
       # Create initial directories
       - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/etc']
       - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/run']
       - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/root']
+      - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/dev']
+      - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/sys']
+      - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/proc']
+      - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/tmp']
+      - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/boot/grub']
       - args: ['mkdir', '-p', '@THIS_COLLECT_DIR@/home/managarm']
```
Comment (ArsenArsen) [inline bootstrap.yml:693]: out of scope for packages

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
I'm not sure how to handle home and such. Maybe making home is fine while we hardcode passwd?

**Comment by avdgrinten** [comment]
We can probably keep hardcoding /home/managarm for now.

**Comment by Dennisbonke** [comment]
As requested, the grub directories are no longer there. As the consensus is that we can continue hardcoding `/home/managarm` for now, this should be good to go now.

**Comment by ArsenArsen** [comment]
sure, but no packages should contain anything in /home, especially considering we'll remove it.
it's better to create/remove this dir when making an image IMHO.

**Comment by Dennisbonke** [comment]
> sure, but no packages should contain anything in /home, especially considering we'll remove it. it's better to create/remove this dir when making an image IMHO.

Eventually we should probably remove that yeah, instead putting common dotfiles in `/etc/skel` and letting `useradd` handle it. For now, this should be fine tho.

**Review by avdgrinten — APPROVED** [review]

**Pull request merged**